reader.go

```diff
@@ -18,7 +18,9 @@ func NewReader(source io.Reader, fromEncoding string, toEncoding string) (*Reade
 	converter, err := NewConverter(fromEncoding, toEncoding)
 
 	if err == nil {
-		return NewReaderFromConverter(source, converter), err
+		reader := iconv.NewReaderFromConverter(source, converter)
+		converter.Close()
```

Comment:
as pointed out in the issue comments, closing the converter here before the reader is used doesn't make much sense. it could make sense to setup a finalizer for the converter here though: https://golang.org/pkg/runtime/#SetFinalizer

I have a feeling that relying on finalizers isn't considered idiomatic go though, which is why i personally leaned towards a new Close function on reader that matches signature of io.Closer - however, using a finalizer means we don't have to track whether the converter was provided to the reader or not (and who's responsible for closing it).